File: java/src/main/java/com/hachicore/demo/jooq/utils/JooqListConditionUtil.java
package com.hachicore.demo.jooq.utils;

import org.jooq.Condition;
import org.jooq.Field;
import org.jooq.impl.DSL;
import org.springframework.util.CollectionUtils;

import java.util.List;

public class JooqListConditionUtil {

    public static <T> Condition inIfNotEmpty(Field<T> idField, List<T> ids) {
        if (CollectionUtils.isEmpty(ids)) {
            return DSL.noCondition();
        }
        return idField.in(ids);
    }

}

File: kotlin/src/main/kotlin/com/hachicore/demo/jooq/utils/JooqListConditionUtil.kt
package com.hachicore.demo.jooq.utils

import org.jooq.Condition
import org.jooq.Field
import org.jooq.impl.DSL

class JooqListConditionUtil {

    companion object {
        fun <T> inIfNotEmpty(idField: Field<T>, ids: List<T>?): Condition {
            return if (ids.isNullOrEmpty()) DSL.noCondition() else idField.`in`(ids)
        }
    }

}